libs/sdk-backend-base/src/workspaceSettingsBackend/index.ts

```diff
@@ -0,0 +1,80 @@
+// (C) 2021 GoodData Corporation
+import identity from "lodash/identity";
+import {
+    IAnalyticalBackend,
+    IUserWorkspaceSettings,
+    IWorkspaceSettings,
+    IWorkspaceSettingsService,
+} from "@gooddata/sdk-backend-spi";
+import { decoratedBackend, WorkspaceSettingsDecoratorFactory } from "../decoratedBackend";
+import { DecoratedWorkspaceSettingsService } from "../decoratedBackend/workspaceSettings";
+
+/**
+ * Adjusts workspace config
+ *
+ * @beta
+ */
+export type SettingsWrapper = (settings: IWorkspaceSettings) => IWorkspaceSettings;
+
+/**
+ * Adjusts current user config
+ *
+ * @beta
+ */
+export type CurrentUserSettingsWrapper = (settings: IUserWorkspaceSettings) => IUserWorkspaceSettings;
+
+class WithModifiedWorkspaceSettingsService extends DecoratedWorkspaceSettingsService {
+    constructor(
+        decorated: IWorkspaceSettingsService,
+        readonly settingsWrapper: SettingsWrapper,
+        readonly currentUserSettingsWrapper: CurrentUserSettingsWrapper,
+    ) {
+        super(decorated);
+    }
+
+    async getSettings(): Promise<IWorkspaceSettings> {
+        const settings = await this.decorated.getSettings();
+        return this.settingsWrapper(settings);
+    }
+
+    async getSettingsForCurrentUser(): Promise<IUserWorkspaceSettings> {
+        const currentUserSettings = await this.decorated.getSettingsForCurrentUser();
+        return this.currentUserSettingsWrapper(currentUserSettings);
+    }
+}
+
+function customWorkspaceSettings(config: WorkspaceSettingsConfiguration): WorkspaceSettingsDecoratorFactory {
+    return (original: IWorkspaceSettingsService) =>
+        new WithModifiedWorkspaceSettingsService(
+            original,
+            config.settingsWrapper || identity,
+            config.currentUserSettingsWrapper || identity,
+        );
+}
+
+/**
+ * Specifies workspace settings and current user settings to be adjusted by the decorator
+ *
+ * @beta
+ */
+export interface WorkspaceSettingsConfiguration {
+    settingsWrapper?: SettingsWrapper;
+    currentUserSettingsWrapper?: CurrentUserSettingsWrapper;
+}
```

Comment:
please describe the settings vs current user settings wrapper & the default behavior.

Reply:
also, I propose to introduce the convenience where it is possible to specify settings for both workspace settings & workspace-user settings in a single place.

the idea is... often time, you are interested in one set of ISettings to be applied in any context (user-workspace, workspace etc). with the current setup, you would have to specify these settings in via two wrappers.

the proposal is, have a single setting that can influence both.